### ISSUE TITLE
Null pointer handling around "PFN_vkFreeFunction" callbacks?

### ISSUE THREAD

**Issue by elfring** [KhronosGroup/Vulkan-Docs#73]
It seems that [the description for the data type "`PFN_vkFreeFunction`"](https://www.khronos.org/registry/vulkan/specs/1.0/apispec.html#VkAllocationCallbacks "Releasing memory") does not mention explicitly so far how null pointers should be treated by the referenced callback function implementation.
How much will it matter if passed null pointers will be tolerated (or not)?

**Comment by courtney-g** [comment]
Under Valid Usage we have:
If pAllocator is not NULL, pAllocator must be a pointer to a valid VkAllocationCallbacks structure

The intent is if you've provided allocators, i.e. pAllocator is not NULL, then all the functions in the VkAllocationCallbacks structure are valid/defined.

Looks like that section needs a little more work.

**Comment by elfring** [comment]
Do you expect that Vulkan memory release implementations will tolerate the passing of null pointers [in the way it is specified for the function "`free`" of the programming language "C"](http://stackoverflow.com/questions/18775608/free-a-null-pointer-anyway-or-check-first "Free a null pointer anyway or check first?")?

**Comment by critsec** [comment]
In the description of PFN_vkFreeFunction, the spec says:

> `pMemory` may be NULL, which the callback must handle safely. If `pMemory` is non-NULL, it must be a pointer previously allocated by pfnAllocation or pfnReallocation and must be freed by the function.

The intent is that `PFN_vkFreeFunction` handles NULL pointers the same was as the C standard library free function.

**Comment by elfring** [comment]
>In the description of PFN_vkFreeFunction, the spec says:

Unfortunately, I do not find the text (from the paragraph you quoted) in [the published document "Vulkan API Reference Pages"](https://www.khronos.org/registry/vulkan/specs/1.0/apispec.html#VkAllocationCallbacks "Reference documentation for the Vulkan API") (according to the commit 7e195e846e9b3d7b91c2fe57406a97288261fc5e on 2016-02-16).
How can the information status be improved?

**Comment by elfring** [comment]
Should [information from the chapter file "memory.txt"](https://github.com/KhronosGroup/Vulkan-Docs/blob/7e195e846e9b3d7b91c2fe57406a97288261fc5e/doc/specs/vulkan/chapters/memory.txt#L113 "Description for PFN_vkFreeFunction") be integrated into any more document formats?

**Comment by courtney-g** [comment]
Changes will be in next 1.0.8 update.

**Comment by elfring** [comment]
Which files were changed for this issue?

**Comment by oddhack** [comment]
Should be fixed in the 1.0.9 spec update.

**Comment by elfring** [comment]
How do you think about to refer to [the following information](https://github.com/KhronosGroup/Vulkan-Docs/blob/5d10fc3089ba0ec6a4c7d757a4b18a9ef215d2fb/doc/specs/vulkan/chapters/memory.txt#L115 "Memory chapter") at more places?

```
pname:pMemory may: be `NULL`, which the callback must: handle safely.
```
